packages/edge-config/src/index.ts

```diff
@@ -85,6 +45,60 @@ interface EdgeConfigClientOptions {
    * This cache is not used in preview or production deployments as superior optimisations are applied there.
    */
   disableDevelopmentCache?: boolean;
+  /**
+   * By default all Edge Config reads will be deduped and cached for duration
+   * of the current request when hosted on Vercel. This ensures that reads of
+   * the same key will return the same value for the duration of a request, and
+   * it also reduces latency as the result will be reused.
+   *
+   * You can disable this by passing `disableRequestContextCache: true` which
+   * will result in Edge Config being read every time.
+   */
+  disableRequestContextCache?: boolean;
+}
+
+interface RequestContextStore {
+  get: () => RequestContext;
+}
+
+interface RequestContext {
+  headers: Record<string, string | undefined>;
+  url: string;
+  waitUntil?: (promise: Promise<unknown>) => void;
+}
+
+type Loaders = ReturnType<typeof createLoaders>;
+
+function getLoadersInstance(
+  options: Parameters<typeof createLoaders>[0] & {
+    disableRequestContextCache?: boolean;
+  },
+  loadersInstanceCache: WeakMap<RequestContext, Loaders>,
+): ReturnType<typeof createLoaders> {
+  if (options.disableRequestContextCache) return createLoaders(options);
+
+  const requestContextStore =
+    // @ts-expect-error -- this is a vercel primitive which might or might not be defined
+    globalThis[Symbol.for('@vercel/request-context')] as
+      | RequestContextStore
+      | undefined;
+
+  const requestContext = requestContextStore?.get();
+
+  // if we have requestContext we can use dataloader to cache and batch per request
+  if (requestContext) {
+    const loadersInstance = loadersInstanceCache.get(requestContext);
+    if (loadersInstance) return loadersInstance;
+
+    const loaders = createLoaders(options);
+    loadersInstanceCache.set(requestContext, loaders);
+    return loaders;
+  }
+
+  // there is no requestConext so we can not cache loader instances per request,
+  // so we return a new instance every time effectively disabling dataloader
+  // batching and caching
+  return createLoaders(options);
```

Comment:
Do you know how much overhead this causes if new instances are created each time?

Reply:
Not sure, but data loader is intended to be used this way, so it should be acceptable

Reply:
I took a few measurements and it's negligible

```
// first call
getLoadersInstance: 0.268ms
// subsequent calls
getLoadersInstance: 0.038ms
getLoadersInstance: 0.065ms
getLoadersInstance: 0.089ms
getLoadersInstance: 0.05ms
getLoadersInstance: 0.027ms
getLoadersInstance: 0.024ms
```